pkgs/tools/networking/network-manager/vpnc/default.nix

```diff
@@ -24,7 +24,8 @@ in stdenv.mkDerivation {
   ];
 
   buildInputs = [ vpnc networkmanager ]
-    ++ stdenv.lib.optionals withGnome [ gtk3 libsecret libnma ];
+    ++ stdenv.lib.optionals withGnome [ gtk3 libsecret libnma ]
+    ++ stdenv.lib.optionals (!withGnome) [ glib ];
```

Comment:
It should be unconditional:

https://gitlab.gnome.org/GNOME/NetworkManager-vpnc/-/blob/e60f4cf06cd79ee1ec913ec277b3d0c2c90184f7/configure.ac#L98